src/core/Definitions.ml

```diff
@@ -360,6 +361,31 @@ let add_stanza_ (st : Stanza.t) self : t =
     in
     let cmd = Misc.str_replace [ "cur_dir", self.cur_dir ] cmd in
 
+    let binary =
+      match binary, inherits_p with
+      | Some b, _ ->
+        if not (str_mem "$binary" cmd) then
+          Error.failf ~loc "Prover's `cmd` does not contain $binary";
+        Misc.str_replace ["cur_dir", self.cur_dir] b
+      | None, Some p ->
+        if str_mem "$binary" cmd then
+          p.binary
+        else
+          Misc.get_binary_of_cmd cmd
```

Comment:
should  this also have `Misc.str_replace ["cur_dir", self.cur_dir]` ?

Reply:
It used to, but I don't think it should, because there is `let cmd = Misc.str_replace [ "cur_dir", self.cur_dir ] cmd in` right above.